src/parser/Parser.cpp

```diff
@@ -183,10 +183,10 @@ void Parser::import(vector<unique_ptr<ImportNode>> &imports) {
     }
 }
 
-// TODO declaration_sequence = [ CONST { const_declaration ";" } ]
-// TODO                        [ TYPE { type_declaration ";" } ]
-// TODO                        [ VAR { variable_declaration ";" } ]
-// TODO                        { procedure_declaration ";" } .
+// declaration_sequence = [ CONST { const_declaration ";" } ]
```

Comment:
The `TODO` here is a reminder to myself that this could be simplified into one production, rather than doing it how it is currently done where there is one main production and four sub-productions (that are never called from anywhere else).

The structure that is currently implemented still follows the grammar of Oberon-0 as described in Wirth's compiler construction book. Wherever the Oberon-0 grammar deviates from the full Oberon-07 grammar, I marked it with `TODO` comments in order to be able to identify these parts of the code quickly, i.e., it doesn't mean that something is not supported. It simply means that the implementation of the parser is structured differently from the grammar and could or should be aligned somewhere down the road.